src/Pool.js

```diff
@@ -264,8 +267,16 @@ Pool.prototype._getWorker = function() {
 Pool.prototype._removeWorker = function(worker) {
   DEBUG_PORT_ALLOCATOR.releasePort(worker.debugPort)
   // terminate the worker (if not already terminated)
-  worker.terminate();
-  this._removeWorkerFromList(worker);
+  return new Promise(function(resolve, reject) {
+    worker.terminate(false, function(err) {
+      me._removeWorkerFromList(worker);
```

Comment:
With the new solution, termination is async, so there is a temporary situation where the worker is still in the worker list but is being terminated. Should we change this to remove the worker from the list first, and after that call .terminate?

Reply:
Are you asking me?  :smile:  I was keeping the series of instructions the same; terminate _then_ remove.  

However, the only place in the codebase that calls `_removeWorker()` removes it synchronously beforehand:

https://github.com/josdejong/workerpool/blob/57e6795ca08d51022847f8de204fca7ae0fb8407/src/Pool.js#L211-L214

It appears the call to `_removeWorkerFromList()` here is not needed unless you expect `_removeWorker()` to be called for some other reason.